File: frontend/src/contract/sseSetupGenerateProgressCode.ts
export const SseSetupGenerateProgressCode = {
  SETUP_SUMMARIZING: 'SETUP_SUMMARIZING',
  SETUP_GENERATING_CHARACTER: 'SETUP_GENERATING_CHARACTER',
  SETUP_PICKING_NAME: 'SETUP_PICKING_NAME',
  SETUP_GENERATING_SCENE: 'SETUP_GENERATING_SCENE',
} as const;

export type SseSetupGenerateProgressCodeValue =
  (typeof SseSetupGenerateProgressCode)[keyof typeof SseSetupGenerateProgressCode];

File: backend/ondgard-chat/src/main/java/com/ondgard/game/chat/contract/sse/campaign/setup/SseSetupGenerateProgressCode.java
package com.ondgard.game.chat.contract.sse.campaign.setup;

import com.ondgard.game.chat.contract.sse.SseProgressValue;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum SseSetupGenerateProgressCode implements SseProgressValue {

  SETUP_SUMMARIZING("SETUP_SUMMARIZING"),
  SETUP_GENERATING_CHARACTER("SETUP_GENERATING_CHARACTER"),
  SETUP_PICKING_NAME("SETUP_PICKING_NAME"),
  SETUP_GENERATING_SCENE("SETUP_GENERATING_SCENE");

  private final String value;
}
